nautobot/ipam/models.py

```diff
@@ -936,7 +937,7 @@ class IPAddress(PrimaryModel, StatusModel, RoleModelMixin):
     objects = BaseManager.from_queryset(IPAddressQuerySet)()
 
     class Meta:
-        ordering = ("ip_version", "host", "prefix_length")  # address may be non-unique
+        ordering = ("parent__namespace", "ip_version", "host", "mask_length")  # address may be non-unique
```

Comment:
Eek cross-table ordering is going to be potentially very expensive at scale. Just something to be aware of.

Reply:
This was my suggestion, I hadn't considered the performance implication. I'd be fine with reverting this as long as we're OK with IPAddress querysets potentially interleaving records from different namespaces?